test/runtest-individual.e2e.js

```diff
@@ -0,0 +1,612 @@
+        return expect(whatsAppTestHeading).toBe('WhatsApp Test')
+      },
+      { timeout: 120000 }
+    )
+
+    await waitFor(
+      async () => {
+        const signalTestHeading = await app.client.getText(
+          'div[data-testid=text-running-test-name]'
+        )
+        return expect(signalTestHeading).toBe('Signal Test')
+      },
+      { timeout: 120000 }
+    )
+
+    await waitFor(
+      async () => {
+        const animationVisible = await app.client.isVisible(
+          'div[data-testid=running-animation-im]'
+        )
+        return expect(animationVisible).toBe(false)
+      },
+      { timeout: 120000 }
+    )
+
+    await app.client.pause(2500)
+  })
+
+  test('Loads Test Results page on completion', async () => {
+    await waitFor(
+      async () =>
+        expect(
+          app.client.isVisible('div[data-testid=overview-tests]')
+        ).resolves.toBe(true),
+      { timeout: 120000 }
+    )
+  })
+})
+
+describe('Websites test', () => {
+  let app
+
+  beforeAll(async () => {
+    app = await startApp()
+  })
+
+  afterAll(async () => {
+    await stopApp(app)
+  })
+
+  test('Website measurement test loads correctly', async () => {
+    await app.client
+      .$('div[data-testid=run-card-websites]')
+      .click()
+      .pause(1000)
+
+    await waitFor(
+      async () =>
+        expect(
+          app.client.isVisible('button[data-testid=button-run-test]')
+        ).resolves.toBe(true),
+      { timeout: 120000 }
+    )
+
+    await screenshotApp(app, 'runtest-description-websites')
+
+    await app.client
+      .$('button[data-testid=button-run-test]')
+      .click()
+      .pause(500)
+
+    const headingTestGroupName = await app.client.getText(
+      'h2[data-testid=heading-test-group-name]'
+    )
+    expect(headingTestGroupName).toBe('Websites')
+
+    const headingPreparingTests = await app.client.getText(
+      'h3[data-testid=heading-running-test-name]'
+    )
+    expect(headingPreparingTests).toBe('Preparing test...')
+  })
+
+  test('Website network runs successfully', async () => {
+    await waitFor(
+      async () => {
+        const runningTestName = await app.client.getText(
+          'div[data-testid=text-running-test-name]'
+        )
+        return expect(runningTestName).toBe('Web Connectivity Test')
+      },
+      { timeout: 300000 }
+    )
+
+    await waitFor(
+      async () => {
+        const headingRunning = await app.client.getText(
+          'h3[data-testid=heading-running-test-name]'
+        )
+        return expect(headingRunning).toBe('Running:')
+      },
+      { timeout: 120000 }
+    )
+
+    await screenshotApp(app, 'runtest-running-websites')
+
+    await waitFor(
+      async () => {
+        const animationVisible = await app.client.isVisible(
+          'div[data-testid=running-animation-websites]'
+        )
+        return expect(animationVisible).toBe(false)
+      },
+      { timeout: 300000 }
+    )
+
+    await app.client.pause(2500)
```

Comment:
Why is there a pause here?

Reply:
This is because in case of a router event, when a new page is loaded, `webdriverIO` keeps looking for matching DOM nodes in the previous page itself. Pausing for a certain amount of time and then performing DOM matching makes `webdriver` look up in the new page.